utils/visualizer.py

```diff
@@ -604,7 +604,7 @@ def create_scatter_plots(
     ):
         """Creates scatter plots for all head predictions."""
         for ihead in range(self.num_heads):
-            if self.head_dims[ihead] // self.num_nodes == 3:
+            if self.head_dims[ihead] == 3:
```

Comment:
Not `> 1`?

Reply:
We were thinking about vectors like magnetic moments. But, sure we can make it more general. Btw, visualization functions still need to be updated after variable graph size.